persistent/Database/Persist/Sql/Run.hs

```diff
@@ -34,15 +35,7 @@ unsafeAcquireSqlConnFromPool
     :: forall backend m
      . (MonadReader (Pool backend) m, BackendCompatible SqlBackend backend)
     => m (Acquire backend)
-unsafeAcquireSqlConnFromPool = do
-    pool <- MonadReader.ask
-
-    let freeConn :: (backend, LocalPool backend) -> ReleaseType -> IO ()
-        freeConn (res, localPool) relType = case relType of
-            ReleaseException -> P.destroyResource pool localPool res
-            _ -> P.putResource localPool res
-
-    return $ fst <$> mkAcquireType (P.takeResource pool) freeConn
+unsafeAcquireSqlConnFromPool = MonadReader.asks poolToAcquire
```

Comment:
The implementation of [`poolToAcquire`](https://hackage.haskell.org/package/resourcet-pool-0.1.0.0/docs/src/Data.Pool.Acquire.html#poolToAcquire) appears to be copied from this function. This function has an `unsafe` prefix, but the function you're copying doesn't have any mention of unsafety in any way.

Looking at the docs, I think the "unsafety" might be coming from the fact that this `Acquire` doesn't start/commit a transaction. So it may be no real concern.

Reply:
Yeah, I mostly factored out this function. I agree, it seems like the unsafety is persistent-wise, not resource-pool-wise.